kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/CsrServiceImpl.java

```diff
@@ -320,6 +326,29 @@ private CallSummaryRecord aggregateSummaryRecord(CallSummaryRecordDto csr) {
     }
 
 
+    private void validateCallSummaryRecord(Subscription subscription, CallSummaryRecordDto csr) throws InvalidCdrData {
+        if (!subscription.getSubscriptionPack().hasMessageWithWeekId(csr.getWeekId())) {
+            throw new InvalidCdrData(String.format("%s is an invalid weekId for pack %s",
+                    csr.getWeekId(), subscription.getSubscriptionPack().getName()));
+        }
+
+        if (!subscription.getSubscriptionPack().hasMessageWithFilename(csr.getContentFileName())) {
```

Comment:
Is there a race condition with updates to the content names?  Just wondering if they ever release new content will this fail?  Not sure it is a likely scenario.  Mostly just curious